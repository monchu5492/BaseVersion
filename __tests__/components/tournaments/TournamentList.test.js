import toJSON from 'enzyme-to-json';
import { shallow, mount } from 'enzyme';
import { compose } from 'redux';
import TournamentList from '../../../src/components/ui/TournamentList';

jest.mock('../../../src/components/ui/Tournament', () =>
  ({ rating, onRate = f => f, onRemove = f => f }) =>
    (<div className="mock-tournament">
      <button className="rate" onClick={() => onRate(rating)} />
      <button className="remove" onClick={onRemove} />
     </div>));

describe('<TournamentList /> UI Component', () => {
  afterAll(() => jest.resetAllMocks());

  describe('Rendering UI', () => {
    it('Renders Correctly', () =>
      compose(expect, toJSON, shallow)(<TournamentList tournaments={_testTournaments} />).toMatchSnapshot());

    it('Defaults Properties correctly', () =>
      expect(shallow(<TournamentList />).find('p').text())
        .toBe('No Tournaments Listed. (Add a Tournament)'));
  });
});
