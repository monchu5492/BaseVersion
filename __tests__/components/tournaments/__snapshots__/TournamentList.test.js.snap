// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TournamentList /> UI Component Rendering UI Renders Correctly 1`] = `
<div
  className="tournament-list"
>
  <Component
    almond_count="16"
    description="This is the first test tourney"
    id="1"
    key="1"
    name="First Tourney"
    owner="1"
    square_count="64"
  />
</div>
`;
